qcodes/instrument_drivers/QuantumDesign/DynaCoolPPMS/DynaCool.py

```diff
@@ -194,11 +212,89 @@ def get_idn(self) -> Dict[str, Optional[str]]:
 
         return dict(zip(('vendor', 'model', 'serial', 'firmware'), idparts))
 
+    def ramp(self, mode: str = "blocking") -> None:
+        """
+        Ramp the field to the value given by the `field_target` parameter
+
+        Args:
+            mode: how to ramp, either "blocking" or "non-blocking". In
+                "blocking" mode, this function does not return until the
+                target field has been reached. In "non-blocking" mode, this
+                function immediately returns.
+        """
+        if mode not in ['blocking', 'non-blocking']:
+            raise ValueError('Invalid ramp mode received. Ramp mode must be '
+                             'either "blocking" or "non-blocking", received '
+                             f'"{mode}"')
+
+        # the target must be converted from T to Oersted
+        target_in_oe = self.field_target()*1e4
+
+        if mode == "blocking":
+            self._do_blocking_ramp(target_in_oe)
+        else:
+            self._field_setter(param='field_setpoint',
+                               value=target_in_oe)
+
+    def _do_blocking_ramp(self, target_in_oe: float) -> None:
+        """
+        Perform a blocking ramp. Only call this function from withing the
+        `ramp` method.
+
+        This method is slow; it waits for the magnet to settle. The waiting is
+        done in two steps, since users have reported that the magnet state does
+        not immediately change to 'ramping' when asked to ramp.
+        """
+
+        target_in_T = target_in_oe*1e-4
+
+        start_field = self.field_measured()
+        ramp_range = np.abs(start_field - target_in_T)
+
+        # TODO (WilliamHPNielsen): do we need some non-zero tolerance here?
```

Comment:
i'd say, yes) some numpy.allclose -like thingy

Reply:
`np.allclose` is surely better than `==`. But we still need some input from scientists of what the noise is.

Reply:
Why do we need to check if it's reached the target value? Is it not enough to read the state from the magnet that it's holding? 
There is certainly some tolerance, but I can't say how much, maybe we can come up with what tolerance is needed? @ThorvaldLarsen  For example when using approach mode "No o-shoot" the field never reaches exactly zero, it has some 0.16 Oe remnant magnetization, however the magnet state says "holding".